core/src/main/resources/lib/form/hetero-list/hetero-list.js

```diff
@@ -10,128 +10,139 @@ Behaviour.specify("DIV.hetero-list-container", 'hetero-list', -100, function(e)
+                        });
+
+                        function o(did) {
+                            if (Object.isElement(did))
+                                did = did.getAttribute("descriptorId");
+                            for (var i = 0; i < templates.length; i++)
+                                if (templates[i].descriptorId == did)
+                                    return i;
+                            return 0; // can't happen
+                        }
 
-                    function o(did) {
-                        if (Object.isElement(did))
-                            did = did.getAttribute("descriptorId");
-                        for (var i=0; i<templates.length; i++)
-                            if (templates[i].descriptorId==did)
-                                return i;
-                        return 0; // can't happen
+                        var bestPos = findBestPosition(t.descriptorId, current, o);
+                        if (bestPos < current.length)
+                            return current[bestPos];
+                        else
+                            return insertionPoint;
                     }
 
-                    var bestPos = findBestPosition(t.descriptorId, current, o);
-                    if (bestPos<current.length)
-                        return current[bestPos];
-                    else
-                        return insertionPoint;
-                }
-                (e.hasClassName("honor-order") ? findInsertionPoint() : insertionPoint).insert({before:nc});
-
-                if(withDragDrop)    prepareDD(nc);
+                    (e.hasClassName("honor-order") ? findInsertionPoint() : insertionPoint).insert({before: nc});
 
-                new YAHOO.util.Anim(nc, {
-                    opacity: { to:1 }
-                }, 0.2, YAHOO.util.Easing.easeIn).animate();
+                    if (withDragDrop) prepareDD(nc);
 
-                Behaviour.applySubtree(nc,true);
-                ensureVisible(nc);
-                layoutUpdateCallback.call();
-            },true);
-        });
-
-        menuButton.getMenu().renderEvent.subscribe(function() {
-            // hook up tooltip for menu items
-            var items = menuButton.getMenu().getItems();
-            for(i=0; i<items.length; i++) {
-                var t = templates[i].tooltip;
-                if(t!=null)
-                    applyTooltip(items[i].element,t);
-            }
-        });
+                    new YAHOO.util.Anim(nc, {
+                        opacity: {to: 1}
+                    }, 0.2, YAHOO.util.Easing.easeIn).animate();
 
-        if (e.hasClassName("one-each")) {
-            // does this container already has a configured instance of the specified descriptor ID?
-            function has(id) {
-                return Prototype.Selector.find(e.childElements(),"DIV.repeated-chunk[descriptorId=\""+id+"\"]")!=null;
-            }
+                    Behaviour.applySubtree(nc, true);
```

Comment:
Can you add a comment to this block to explain its intent?

Reply:
I have no idea it's intent, I added an if block to safe guard against an undefined value, the rest of the changes are all indentation, you can see the changes without white space related changes showing my using this url:
https://github.com/jenkinsci/jenkins/pull/4479/files?utf8=%E2%9C%93&diff=split&w=1